src/main/java/herrekt/Herrekt.java

```diff
@@ -0,0 +1,89 @@
+package herrekt;
+
+import herrekt.exceptions.InvalidInputException;
+import herrekt.exceptions.NoTaskException;
+import herrekt.taskmanager.*;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+
+public class Herrekt {
+
+    private final Ui ui;
+    private final Storage storage;
+    private TaskList tasks;
+    private final Parser parser;
+
+    public Herrekt(String filePath) {
+        this.parser = new Parser();
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            this.tasks = new TaskList(this.storage.load());
+        } catch (FileNotFoundException e) {
+            this.ui.printLoadingError();
+            this.tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        this.ui.printWelcomeMessage();
+        Scanner sc = new Scanner(System.in);
+
+        String phrase = sc.nextLine();
+
+        while (!phrase.equals("bye")) {
+            try {
+                if (phrase.equals("list")) {
+                    ui.printTaskList(tasks);
+                } else if (phrase.contains("done")) {
+                    int taskNumber = parser.parseDoneInputToInt(phrase);
+                    tasks.markAsDone(taskNumber);
+                    ui.printMarkTaskAsDone(taskNumber, tasks);
+                } else if (phrase.contains("delete")) {
+                    int taskNumber = parser.parseDeleteInputToInt(phrase);
+                    ui.printTaskDeleted(taskNumber, tasks);
+                    tasks.delete(taskNumber);
+                } else {
+                    isInputValid(phrase);
+                    Task task = parser.parsePhraseToTask(phrase);
+                    tasks.add(task);
+                    ui.printNumberOfTasks(tasks);
+                }
+            } catch (ArrayIndexOutOfBoundsException e3) {
+                ui.printIncorrectFormatError(phrase);
+            } catch (IndexOutOfBoundsException e) {
+                ui.printInputBiggerThanTaskList(tasks);
+            } catch (InvalidInputException e1) {
+                ui.printInvalidInputError(phrase);
+            } catch (NoTaskException e2) {
+                ui.printNoTaskError(phrase);
+            }
```

Comment:
The `catch` statements helps to catch the invalid inputs. How about  including an `if` condition to  check for valid phrases at the beginning? 

And then an `else` block to handle the invalid phrases in a separate method? 

This will ensure SLAP and increase the readability of your code.